internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -723,20 +723,29 @@ func (r *NumaflowControllerReconciler) ErrorHandler(numaflowController *apiv1.Nu
 
 // areDependentResourcesDeleted checks if dependent resources are deleted.
 func (r *NumaflowControllerReconciler) areDependentResourcesDeleted(ctx context.Context, controller *apiv1.NumaflowController) bool {
-	pipelineList, err := kubernetes.ListLiveResource(ctx, common.NumaflowAPIGroup, common.NumaflowAPIVersion, "pipelines",
+	pipelineList, err := kubernetes.ListLiveResource(ctx, common.NumaflowAPIGroup, common.NumaflowAPIVersion, common.NumaflowPipelineKind,
 		controller.Namespace, common.LabelKeyParentRollout, "")
 	if err != nil {
-		return false
+		if !apierrors.IsNotFound(err) {
+			return false
```

Comment:
so, this is the case of getting a real error when listing? Should we maybe return an error to `Reconcile()` if that happens so that we'll re-reconcile?

Reply:
Anyway, I'm trying to understand how critical the bug is that you're fixing here. It seems like previously in the case of `apierrors.IsNotFound()`, then we would return false. In that case, we must pretty much never be deleting the `NumaflowController` CR - is that right?

Reply:
1. Yes, error can be used as well to reconcile.
2. Yes this was the bug that Numaflow controller CR was never get's deleted, I'm not sure why it didn't cause the problem till now